### PR TITLE
Make buttons work on Work screen and various other UI things

### LINKED ISSUE #32: Add Temperature Presets for common fast Temperature changes
would be nice to have 2-3 common presets for  `Set °C`
quickly tap on for the change.
I often go from Lead solder to a no-lead solder and would love to be lazy and not even scroll and pick from 3 different ones which would probably take care of most use cases.

2- 3 boxes is good.  of course still keep the custom change feature but being able to save 3 numbers locally is nice. I suppose we would need a text file saved into the same folder as PineSAM to keep user defined settings that are accessible by the script.

### PR DESCRIPTION
**Pull request by NeilHanlon**
Work screen improvements and functionality implmementation

Closes #32

- Make plus/minus buttons control set temperature as per Short/Long
  press parameters. Tapping the buttons once change the set point by
  TempChangeShortStep. Holding them changes the set point by
  TempChangeLongStep
- Work Screen changed to add preset buttons which immediately set the
  iron to the specified temperature. Holding down this button will
  prompt to confirm setting a new preset value using the current
  SetPoint.
- Presets are stored in localstorage in the browser
- Peak wattage is displayed alongside other live data
- display proper temperature based on setting

### CODE REVIEW

**Comment by NeilHanlon** [comment]
Ugh. I'm sorry my linter made a mess of the javascript diff for this PR :(

**Comment by River-Mochi** [comment]
User testing issue.
1. needs a longer delay on buttons
2. when I click it one time with mouse sometimes I get 2-3 degrees jumps. (even though I have Short set to "1 in soldering settings" sometimes if I'm fast enough it's just 1 increment but most of the time it's too fast and jumps 2-3 degrees.
3. test on phone too, when I tap with finger, it's too fast and triggering 2-3 degree jumps instead of just "1"  it thinks I'm tapping 2-3 times and not just 1 time bc my finger and mouse can't unclick /pull back fast enough.

**Comment by builder555** [comment]
@NeilHanlon Thanks for the pr. The UI is getting pretty busy, i'm thinking of moving it to a proper vue.js project to break it up into components. Once you address my comments, I can merge this and then make a GH action to do proper UI builds so we don't end up with a 5000 line html and 8000 line js 😅

**Comment by NeilHanlon** [comment]
> @NeilHanlon Thanks for the pr. The UI is getting pretty busy, i'm thinking of moving it to a proper vue.js project to break it up into components. Once you address my comments, I can merge this and then make a GH action to do proper UI builds so we don't end up with a 5000 line html and 8000 line js sweat_smile

Yep, makes sense to me. I'm happy to rebase my stuff on top if you want to reorganize first.

I want to squash a couple bugs in this before it goes in, anyways (Mostly the UX things River mentioned)

**Comment by NeilHanlon** [comment]
@builder555 im not seeing your comments on my side. I think you may have to click the "publish review" button in the review screen

**Comment by builder555** [comment]
> @builder555 im not seeing your comments on my side. I think you may have to click the "publish review" button in the review screen

that's my mistake, i added them but forgot to click "submit review"

File: ui/settings.html [hunk 1]
```diff
@@ -61,17 +61,30 @@ <h1 class="title" style="pointer-events: none!important;">
...
-          <i class="fa fa-bolt" style="font-size:0.6em; font-weight:600;"></i> {{liveData.Watts}} W
+          <div style="font-size:0.2em; margin-top:-1em">CURRENT TEMPERATURE</div>
         </div>
-        <div class="live-power">
-          <i class="fa fa-plug" style="font-size:0.6em; font-weight:600;"></i> {{liveData.Voltage}} V
+        <div class="info-row">
+          <div class="row-item">
+              <div>
+                <i class="fa-solid fa-gauge-high"></i> {{liveData.Watts}} W 
+                <span class="peak" style="font-size: 0.8em; display: block; margin-top: -0.5em;">peak</span>
```
[the remark this answers is not in the document]
Comment (River-Mochi) [reply ui/settings.html]: Yes even with graph having box to show peak watts is good bc you can't watch graph at every second while also soldering . This is why on joric graph I asked for specific display of whatever peak is in whole session. Also we can't record the graph now either so peak box is only reference. 
Yes needed for ease to find it even if there is graph.

File: ui/settings.html [hunk 2]
```diff
@@ -61,17 +61,30 @@ <h1 class="title" style="pointer-events: none!important;">
...
-        <div class="live-power">
-          <i class="fa fa-bolt" style="font-size:0.6em; font-weight:600;"></i> {{liveData.Watts}} W
+          <div style="font-size:0.2em; margin-top:-1em">CURRENT TEMPERATURE</div>
         </div>
-        <div class="live-power">
-          <i class="fa fa-plug" style="font-size:0.6em; font-weight:600;"></i> {{liveData.Voltage}} V
+        <div class="info-row">
+          <div class="row-item">
+              <div>
+                <i class="fa-solid fa-gauge-high"></i> {{liveData.Watts}} W
```
[the remark this answers is not in the document]
Comment (NeilHanlon) [reply ui/settings.html]: This is fixed in the latest commit

File: ui/settings.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (NeilHanlon) [reply ui/settings.html]: Latest commit should bring these all in line, as well as remove the 'peak' word.

[the remark this answers is not in the document]
Comment (NeilHanlon) [reply ui/settings.html]: Will leave this open re: if peak wattage is needed. IMO, it's nice to have, but not 'needed' per se. That said, I do _like_ having it.

File: ui/settings.html [hunk 3]
```diff
@@ -61,17 +61,30 @@ <h1 class="title" style="pointer-events: none!important;">
     <div class="is-flex is-justify-content-center mb-5">
       <div class="temp-holder" style="pointer-events: auto;">
         <div class="live-temp">
-          <button class="button change minus"> <i class="fas fa-minus"></i> </button>
-          <button class="button change plus"> <i class="fas fa-plus"></i> </button>
-          <div style="font-size:0.3em; margin-top:-1.8em; margin-bottom:0.8em">SET: {{settings.SetTemperature.value}}&deg;C</div>
+          <div class="buttons are-large">
+            <button class="button change minus" @click="setTemperature(-1, $event)" @touchstart="setTemperature(-1, $event)" @mousedown="setTemperature(-1, $event)" @touchleave="setTemperature(0, $event)" @mouseleave="setTemperature(0, $event)" @mouseup="setTemperature(0, $event)" @touchend="setTemperature(0, $event)"> <i class="fas fa-minus"></i> </button>
```
[the remark this answers is not in the document]
Comment (NeilHanlon) [reply ui/settings.html:65]: as River mentioned, the latest commit has the long/short press matching that of the pinecil better and works on desktop+mobile.

from a UI perspective, I can see how it's not very accessible. Perhaps that is something we can iterate on later?

[the remark this answers is not in the document]
Comment (River-Mochi) [reply ui/settings.html:65]: > from a UI perspective, I can see how it's not very accessible.

so far it's amazingly  working right now after the fixes done, on both iphone and Android for long button presses to increment temp on +/-  
so  I think keep them until something comes up?

**Pull request merged**